src/connection/server_revive.cpp

```diff
@@ -0,0 +1,165 @@
+/*
+ *
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+//
+// server_revive.cpp - Server Revive monitoring module
+//
+
+#include "server_revive.hpp"
+#include "util_func.h"
+
+master_monitor::master_monitor()
+{
+  m_monitoring_thread = std::make_unique<std::thread> ([this]()
+  {
+    master_monitor_worker();
+  });
+}
+
+
+// When master_monitor is deleted, it should guerentee that
+// monitoring_thread is terminated before monitor_list is deleted.
+master_monitor::~master_monitor()
+{
+  if (m_monitoring_thread->joinable())
+    {
+      m_monitoring_thread->join();
+    }
+  m_monitor_list.reset();
+}
+
+void
+master_monitor::make_and_insert_server_entry (int pid, const char *server_name, const char *exec_path, char *args,
+    CSS_CONN_ENTRY *conn)
+{
+  auto temp = std::make_unique<server_entry> (pid, server_name, exec_path, args, conn);
+  m_monitor_list->server_entry_list.push_back (std::move (temp));
+}
+
+void
+master_monitor::master_monitor_worker (void)
+{
+  while (true)
+    {
+      // TODO: Server_entry selection which need_revive value is true. (Will be implemented in CBRD-25438 issue.)
+    }
+}
+
+void master_monitor::remove_server_entry (std::unique_ptr<server_entry> entry)
+{
+  auto it = std::find_if (m_monitor_list->server_entry_list.begin(), m_monitor_list->server_entry_list.end(),
+			  [&entry] (const std::unique_ptr<server_entry> &e)
+  {
+    return e.get() == entry.get();
+  });
+
+  if (it != m_monitor_list->server_entry_list.end())
+    {
+      m_monitor_list->server_entry_list.erase (it);
+    }
+}
+
+std::unique_ptr<master_monitor::server_entry> master_monitor::get_server_entry_by_conn (CSS_CONN_ENTRY *conn)
+{
+  auto it = std::find_if (m_monitor_list->server_entry_list.begin(), m_monitor_list->server_entry_list.end(),
+			  [conn] (const std::unique_ptr<server_entry> &e)
+  {
+    return e->conn == conn;
+  });
+
+  if (it != m_monitor_list->server_entry_list.end())
+    {
+      return std::move (*it);
+    }
+
+  return nullptr;
+}
+
+void
+master_monitor::revive_server (std::unique_ptr<server_entry> entry)
+{
+  // TODO: Server reviving routine for abnormal terminated server. (Will be mplemented in CBRD-25438 issue.)
+}
+
+master_monitor::server_entry::
+server_entry (int pid, const char *server_name, const char *exec_path, char *args, CSS_CONN_ENTRY *conn)
```

Comment:
```suggestion
server_entry (int pid, const char *server_name, const char *exec_path, char *args, CSS_CONN_ENTRY *conn)
: m_pid {pid}
, m_server_name {server_name}
, m_exec_path {exec_path}
, m_conn {conn}
```

이런식으로 초기화되면 좋겠네요.
member 변수들의 네이밍도 저희 C++ 컨벤션에 맞추면 좋을 듯 하구요. 
m_server_name이나, m_exec_path도 const std::string 데이터형으로 선언하시면, 자동으로 char* server_name, char* exec_path 받아와서 초기화됩니다. 
나머지 변수들은, 아래 구현부에서 초기화해주면 되겠네요.

Reply:
수정하였습니다.